### DISCUSSION TITLE
Error handling upgrade request TypeError: Cannot read properties of undefined (reading 'bind')

### DISCUSSION THREAD

**Discussion by Alisadaintanvir** [vercel/next.js#68716] [App Router]
This error occurs while adding matcher in my middleware.

`import { authConfig } from "./auth.config";
import NextAuth from "next-auth";

export const { auth: middleware } = NextAuth(authConfig);

export const config = {
  matcher: ["/((?!api|_next/static|_next/image|favicon.ico).*)"],
};
`

what to do??
I am using next js latest version

**Comment by ananuness** [comment]
Hi!! As you want to do a regex, you can remove those quotes, scape all foward slashes, the dot in `.ico`, also adding scaped foward slashes to the other alternatives and it will be like:

```js
export const config = {
  matcher: [/\/(?!api|\/_next\/static|\/_next\/image|\/favicon\.ico).*/],
};
```
